qiskit_neko/tests/machine_learning/test_neural_network_classifier_primitives.py

```diff
@@ -0,0 +1,55 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for quantum neural networks classifier."""
+import numpy as np
+from ddt import ddt, data
+from qiskit.algorithms.optimizers import COBYLA
+from qiskit.primitives import Sampler as ReferenceSampler
+from qiskit.utils import algorithm_globals
+from qiskit_aer.primitives import Sampler as AerSampler
+from qiskit_machine_learning.algorithms.classifiers import VQC
+
+from qiskit_neko import decorators
+from qiskit_neko.tests import base
+
+
+@ddt
+class TestNeuralNetworkClassifierOnPrimitives(base.BaseTestCase):
+    """Test adapted from the qiskit_machine_learning tutorials."""
+
+    def setUp(self):
+        super().setUp()
+        if hasattr(self.backend.options, "seed_simulator"):
+            self.backend.set_options(seed_simulator=42)
+
+        self.samplers = dict(reference=ReferenceSampler(), aer=AerSampler())
+
+    @decorators.component_attr("terra", "backend", "machine_learning")
```

Comment:
Since this doesn't actually use a user specified backend it's either the terra reference implementation or aer's primitives I think we should just mark this as aer instead of backend:

```suggestion
    @decorators.component_attr("terra", "aer", "machine_learning")
```